### PR TITLE
Add support for params passed indirectly

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit adds support for the `MAX_FLAT_PARAMS` variable in the
current canonical ABI draft. Argument lists which exceed a static number
of parameters are now always passed indirectly through memory. In export
this is dynamically allocated/freed and in imports this is statically
passed.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Note that the first commit here is https://github.com/bytecodealliance/wit-bindgen/pull/207

File: crates/gen-spidermonkey/src/lib.rs [hunk 1]
```diff
@@ -48,191 +48,191 @@ lazy_static! {
             WasmSignature {
                 params: vec![],
                 results: vec![],
-                retptr: false,
+                retptr: false,indirect_params:false,
```
Comment (peterhuene) [inline crates/gen-spidermonkey/src/lib.rs:51]: Looks like this file has some bad formatting, probably from conflict resolution?

File: crates/parser/src/abi.rs [hunk 2]
```diff
@@ -654,6 +665,19 @@ def_instruction! {
             params: usize,
         } : [*params + 2] => [0],
 
+        /// TODO
```
Comment (peterhuene) [inline crates/parser/src/abi.rs:668]: Description needed.

File: crates/parser/src/abi.rs [hunk 3]
```diff
@@ -654,6 +665,19 @@ def_instruction! {
...
+        Malloc {
+            realloc: &'static str,
+            size: usize,
+            align: usize,
+        } : [0] => [1],
+
+        /// TODO
```
Comment (peterhuene) [inline crates/parser/src/abi.rs:675]: Description needed.

**Review by peterhuene — APPROVED** [review]
Also looks good, just a few minor comments.

File: crates/gen-spidermonkey/src/lib.rs (hunk 1 above)
Comment (alexcrichton) [reply crates/gen-spidermonkey/src/lib.rs]: Oh this is me using a random macro and praying rustfmt will fix everything for me when, in fact, it did not.

**Pull request merged**